templates/site/js/main.js

```diff
@@ -114,129 +49,159 @@ function login(){
                 password: passwordInput,
                 provider_name:"ziggurat"
                 })
-            }).then(response => response.json()).then(json  => {
+            }).then(response => response.json().then(json  => {
 
             try{
                 if(json.code == 200){
-
                     window.location.href = accountHome;
-                    //console.log(json);
                 }
                 else{
                     if(loginErrorMessage.classList.contains("display-none")){
-                        loginErrorMessage.classList.toggle("display-flex");
+                        loginErrorMessage.classList.toggle("display-none");
                     }
                 }
 
             }catch (error){
-                 //throw new Error(error)
                 console.log(error);
 
             }
-        })
+        }))
     }
 }
 
-function getAuthTkt(response){
-    let authTkt;
-    var json = response.json();
-
-    try{
+function getSessionDetails(){
+    const sessionURLFragment = "/magpie/session";
 
-        if(json.code == 200){
+    return fetch(sessionURLFragment,{
+        method: "GET",
+        headers: {
+            "Accept": "application/json, text/plain",
+            "Content-Type": "application/json"
+            }
+        }).then(response => {return response.json()})
+}
 
-            var cookieArray = response.headers.getSetCookie();
+//User functions
+//--------------
+function checkUserAuthenticated(){
+    getSessionDetails().then(json => {
+        if(json.code && json.code == 200) {
+            if (json.authenticated === false) {
+                window.location.href = loginHome;
+            }
+        }
+    })
+}
 
-            if(cookieArray.length > 0){
-                authTkt = cookieArray[0];
-                return authTkt;
+function getUserDetails(){
+    const loggedUserURLFragment = "/magpie/users/current";
+    return fetch(loggedUserURLFragment,{
+        method: "GET",
+        headers: {
+            "Accept": "application/json, text/plain",
+            "Content-Type": "application/json"
             }
+        }).then(response => {return response.json()})
+}
+
+function setHiddenFields(){
+    var hiddenUsername = document.getElementById("hiddenUsername");
+    var hiddenEmail = document.getElementById("hiddenEmail");
+
+        getUserDetails().then(json => {
+        hiddenUsername.value = json.user["user_name"];
+        hiddenEmail.value = json.user["email"];
+    })
+}
+
+function setUserAccountDetails(setUserDetailsFunction){
+    const targetNode = document.getElementById("hiddenUsername");
+    const config = { attributes: true};
+
+    const callback = (mutationList, observer) => {
+      for (const mutation of mutationList) {
+        if (mutation.type === "attributes") {
```

Comment:
the if and the else block do the same thing?

Reply:
Forgot the API version of the set value functions.

Added the stop observer command.